GenderPayGap.WebUI/Controllers/Admin/AdminDownloadsController.cs

```diff
@@ -0,0 +1,434 @@
+                .Where(org => org.LatestReturn == null || org.LatestReturn.IsLateSubmission)
+                .Include(org => org.Returns)
+                .ToList();
+
+            var records = organisationsWithLateReturns.Select(
+                    org => new
+                    {
+                        org.OrganisationId,
+                        org.OrganisationName,
+                        org.SectorType,
+                        Submitted = org.LatestReturn != null,
+                        ReportingDeadline = org.SectorType.GetAccountingStartDate().AddYears(1).AddDays(-1).ToString("d MMMM yyyy"),
+                        SubmittedDate = org.LatestReturn?.Created,
+                        ModifiedDate = org.LatestReturn?.Modified,
+                        org.LatestReturn?.LateReason
+
+                    })
+                .ToList();
+
+            string fileDownloadName = $"Gpg-LateSubmissions-{DateTime.Now:yyyy-MM-dd HH:mm}.csv";
+            FileContentResult fileContentResult = CreateCsvDownload(records, fileDownloadName);
+
+            return fileContentResult;
+        }
+        
+        [HttpGet("downloads/all-users")]
+        public FileContentResult DownloadAllUsers()
+        {
+            List<User> users = dataRepository.GetAll<User>().ToList();
+
+            var records = users.Select(
+                    u => new
+                    {
+                        u.UserId,
+                        u.Firstname,
+                        u.Lastname,
+                        u.JobTitle,
+                        u.EmailAddress,
+                        u.EmailVerifySendDate,
+                        u.EmailVerifiedDate,
+                        u.Status,
+                        u.StatusDate,
+                        u.StatusDetails,
+                        u.Created
+                    })
+                .ToList();
+
+            string fileDownloadName = $"Gpg-AllUsers-{DateTime.Now:yyyy-MM-dd HH:mm}.csv";
+            FileContentResult fileContentResult = CreateCsvDownload(records, fileDownloadName);
+
+            return fileContentResult;
+        }
+        
+        [HttpGet("downloads/user-organisation-registrations")]
+        public FileContentResult DownloadUserOrganisationRegistrations()
+        {
+            List<UserOrganisation> userOrganisations = dataRepository.GetAll<UserOrganisation>()
+                .Include(uo => uo.Organisation.LatestScope)
+                .Include(uo => uo.User)
+                .ToList();
+
+            var records = userOrganisations.Select(
+                    uo => new
+                    {
+                        uo.OrganisationId,
+                        uo.UserId,
+                        uo.Organisation.OrganisationName,
+                        uo.Organisation.CompanyNumber,
+                        uo.Organisation.SectorType,
+                        uo.Method,
+                        uo.Organisation.LatestScope.ScopeStatus,
```

Comment:
Is there always a `LatestScope`.
I could imagine this might be null sometimes, which would cause a NullReferenceException.
I wonder if we even need the scope status - I'd be happy to remove it

Reply:
Good point. I've removed this from both userOrganisation downloads